src/main/java/org/terasology/taskSystem/actions/AddCropTransferToStorageTaskNode.java

```diff
@@ -0,0 +1,155 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.taskSystem.actions;
+
+import org.terasology.Constants;
+import org.terasology.buildings.components.ConstructedBuildingComponent;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.holdingSystem.components.HoldingComponent;
+import org.terasology.logic.behavior.BehaviorAction;
+import org.terasology.logic.behavior.core.Actor;
+import org.terasology.logic.behavior.core.BaseAction;
+import org.terasology.logic.behavior.core.BehaviorState;
+import org.terasology.logic.common.DisplayNameComponent;
+import org.terasology.math.Region3i;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.network.ColorComponent;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.Color;
+import org.terasology.spawning.OreonSpawnComponent;
+import org.terasology.taskSystem.AssignedTaskType;
+import org.terasology.taskSystem.BuildingType;
+import org.terasology.taskSystem.DelayedNotificationSystem;
+import org.terasology.taskSystem.components.TaskComponent;
+import org.terasology.taskSystem.tasks.HarvestTask;
+import org.terasology.taskSystem.tasks.PlaceBlocksInStorageTask;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockComponent;
+import org.terasology.world.block.BlockManager;
+
+import java.util.List;
+
+@BehaviorAction(name = "add_crop_transfer_to_storage_task")
+public class AddCropTransferToStorageTaskNode extends BaseAction {
+
+    @In
+    private BlockManager blockManager;
+
+    @In
+    private Context context;
+
+    @In
+    private EntityManager entityManager;
+
+    private BlockEntityRegistry blockEntityRegistry;
+    private EntityRef notificationMessageEntity;
+    private DelayedNotificationSystem  delayedNotificationSystem;
+
+    @Override
+    public void construct(Actor oreon) {
+        blockEntityRegistry = context.get(BlockEntityRegistry.class);
+        delayedNotificationSystem = context.get(DelayedNotificationSystem.class);
+
+        notificationMessageEntity = entityManager.create(Constants.NOTIFICATION_MESSAGE_PREFAB);
+
+        DisplayNameComponent displayNameComponent = notificationMessageEntity.getComponent(DisplayNameComponent.class);
+        displayNameComponent.name = "Oreons";
+
+        ColorComponent colorComponent = notificationMessageEntity.getComponent(ColorComponent.class);
+        colorComponent.color = Color.BLACK;
+
+        notificationMessageEntity.saveComponent(displayNameComponent);
+        notificationMessageEntity.saveComponent(colorComponent);
+    }
+
+    @Override
+    public BehaviorState modify(Actor oreon, BehaviorState result) {
+        TaskComponent oreonTaskComponent = oreon.getComponent(TaskComponent.class);
```

Comment:
This is a somewhat long'ish method with unclear grouping / newlines - maybe group into clear blocks and add a comment to each explaining the goal?